### ISSUE TITLE
mainmenu: Wrong position of menu and submenus on Wayland

### ISSUE THREAD

**Issue by gen2brain** [lxqt/lxqt-panel#2171]
##### Expected Behavior

The menu and submenus should be shown in the correct positions.

##### Current Behavior

The attached screenshot shows the issue.

##### Steps to Reproduce (for bugs)

Start Wayland session with `labwc` compositor.

##### Context

There are no issues with the fancymenu, but I prefer to use plain ol' mainmenu even with this issue.

##### System Information

* LXQt Version: 2.1.0
* LXQt panel Version: 2.1.1
* Distribution & Version: Gentoo
* Qt Version: 6.7.3
* liblxqt Version: 2.1.0

![2024-11-16_09-05](https://github.com/user-attachments/assets/7f29464a-6b6f-4350-8e55-be0245a3c910)

**Comment by marcusbritanicus** [comment]
I can confirm this on Wayfire as well with 2.1.0

![2024-11-16_14:00:49](https://github.com/user-attachments/assets/fb4f27f6-16e4-4ad4-8df3-df7c2e9509af)

**Comment by tsujan** [comment]
Thanks, but we've been aware of this for a long time (there may be an open report about it). It's a Qt bug.

I you use Main Menu, put the panel at the top.

**Comment by gen2brain** [comment]
I only found this one https://github.com/lxqt/lxqt-panel/issues/1990, but it is closed, marked as wontfix, and didn't look like my case at first. I guess I will have to leave with it in the Wayland session, I don't use it often anyway. Is there an open Qt bug that I can follow?

**Comment by tsujan** [comment]
> I only found this one https://github.com/lxqt/lxqt-panel/issues/1990,

Yes, I wasn't sure that we had a report about it; hence not closing this one. It's good to keep track of such Qt problems. Main Menu should be OK if you put the panel at the top.

> Is there an open Qt bug that I can follow?

I'm not aware of any. By Qt, I meant `layer-shell-qt` or QtWayland (or both); Qt's base (`qt6-base`) may be OK.

**Comment by gen2brain** [comment]
Thanks. I may as well try to use the new and fancy menu. I have decades of using the bottom panel everywhere, and I have used your panel since razor-qt, habits are hard to break.

**Comment by tsujan** [comment]
You might see the same problem in some other menus, especially in context menus (Status Notifier?) with a bottom/right panel.

We usually don't wait for a fix in Qt but make workarounds. Sadly, no workaround is found for this one yet.

**Comment by stefonarch** [comment]
The same can be seen with the "add|remove to favorites" dialog in fancy menu and with directorymenu widget, only with right or bottom panels. Also tooltips cover the items with those positions. So top panel on wayland  or living with it. 
I've seen a Qt bug about it but can't find it atm.

**Comment by q4a** [comment]
Hi. Is there news about upstream fix of this bug?
Will [Wayland taskbar support v2](https://github.com/lxqt/lxqt-panel/pull/2043) help with this issue?

**Comment by tsujan** [comment]
@q4a 
Thanks for your informative comment!

**Comment by q4a** [comment]
Thanks to several pings and even [one small MR](https://codereview.qt-project.org/c/qt/qtbase/+/681757) we got much bigger MR started in 2023 and it should fix issue:
https://codereview.qt-project.org/c/qt/qtbase/+/491545

If I got it right: Qt 6.10 was released 2 days ago and we got this fix in Qt 6.11:
https://wiki.qt.io/Qt_6.11_Release

**Comment by elviosak** [comment]
In labwc no changes with 6.11 either, and tooltips seems to have the same problem now (AFAIK it didn't on 6.10).
